### PR TITLE
metal: Implement ConstantBuffers

### PR DESCRIPTION
**Pull request by scoopr**

### CODE REVIEW

File: src/backend/metal/src/command.rs [hunk 1]
```diff
@@ -599,6 +599,17 @@ impl RawCommandBuffer<Backend> for CommandBuffer {
                                         }
                                     }
                                 }
+                                ConstantBuffer(ref buffers) => {
+                                    if let EncoderState::Render(ref encoder) = inner.encoder_state {
```
Comment (kvark) [inline src/backend/metal/src/command.rs]: looks like we'll miss a call to `add_buffers` if the encoder is not recording. We should call it before checking the encoder

File: src/backend/metal/src/command.rs [hunk 2]
```diff
@@ -599,6 +599,17 @@ impl RawCommandBuffer<Backend> for CommandBuffer {
...
+                                        for (i, ref bref) in buffers.iter().enumerate() {
+                                            if let Some(ref buffer) = **bref {
+                                                let offset = 0; // TODO: ?
```
Comment (kvark) [inline src/backend/metal/src/command.rs]: should be stored in the `DescriptorSetBinding` alongside the buffer ID

File: src/backend/metal/src/command.rs (hunk 1 above)
Comment (scoopr) [reply src/backend/metal/src/command.rs]: Ah so thats why it was done like that above, I thought it wouldn't matter then.

Comment (kvark) [reply src/backend/metal/src/command.rs]: I haven't double-checked if it matters, tbh, assumed it does

File: src/backend/metal/src/command.rs (hunk 2 above)
Comment (scoopr) [reply src/backend/metal/src/command.rs]: Oh right, in the ConstantBuffer variant itself? Alright then

Comment (kvark) [reply src/backend/metal/src/command.rs]: Basically, store it besides the buffer. Buffer + offset is like a constant buffer view in D3D12

**Comment by scoopr** [comment]
New attempt

**Pull request merged**